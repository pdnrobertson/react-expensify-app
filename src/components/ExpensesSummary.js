import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {

    calculateExpensesTotal = () => {
        return expensesTotal(this.props.expenses)/100
    }

    render() {
        return (
            <div>
                <p>Expenses Total: {numeral(this.calculateExpensesTotal()).format('$0,0,00')}</p>
                <p>Number of Expenses: {this.props.expenses.length}</p>
            </div>
        )
    }
 }

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);