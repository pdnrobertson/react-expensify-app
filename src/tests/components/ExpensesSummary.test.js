import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
})

test('Should render ExpensesSummary with multiply expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});