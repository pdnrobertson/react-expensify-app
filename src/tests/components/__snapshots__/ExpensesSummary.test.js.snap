// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
     
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with multiply expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2.35
     
  </h1>
</div>
`;
